src/widget/wfindonmenu.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <QAction>
+#include <QMenu>
+#include <QUrlQuery>
+
+class Track;
+
+class WFindOnMenu : public QMenu {
+    Q_OBJECT
+  public:
+    explicit WFindOnMenu(
+            QWidget* parent = nullptr);
+    ~WFindOnMenu() override = default;
+
+    enum class Service {
+        SoundCloud,
+        LastFm,
+        Discogs
+    };
+
+    enum class TrackProperty {
+        Artist,
+        ArtistTitle,
+        ArtistAlbum,
```

Comment:
This terminology is confusing, because there actually is a property that is named "album artist". If you intend to mean `ArtistAndAlbum` then please be explicit about it.

If the enum variants denote a conjunction of one or more individual properties then `TrackSearchProperties` would be a better name.

Reply:
For ArtistTitle and ArtistAlbum, I wasn't sure to use "and". Next time I will be more explicit and precise. I will make them "ArtistAndAlbum", "ArtistAndTitle" 

Right now I am fully aware about the enum naming. If they denote single property, they should be named Singular and if they denote conjunction, they should be named Plural. Thanks for this review! I will make it "TrackSearchProperties" as you suggested.